### ISSUE TITLE
List rust-src as a dependency in docs

### ISSUE THREAD

**Issue by Koxiaet** [rust-analyzer/rust-analyzer#3156]
# Bug

In front of enums and structs, `rust-analyzer` claims they have _`0 implementations`_, despite them having implementations:

![zero-impl](https://user-images.githubusercontent.com/38139193/74591849-5e9a6980-5013-11ea-9cc1-0df8bbca6863.png)

# Configuration

I'm using `ra_lsp_server` `58f4dcf` (latest as of now). I installed it with `cargo install --git https://github.com/rust-analyzer/rust-analyzer ra_lsp_server`. I'm using Neovim, with my init.vim:

```vim
call plug#begin()
Plug 'autozimu/LanguageClient-neovim', {
    \ 'branch': 'next',
    \ 'do': 'bash install.sh',
    \ }
let g:LanguageClient_serverCommands = {
\ 'rust': ['ra_lsp_server'],
\ }
call plug#end()
```

`checkhealth` is completely fine, although has an outdated pynvim (0.3.2), no ruby and no node-js.

**Comment by bjorn3** [comment]
Do you have an `Cargo.toml` file in the directory you started VIM from?

**Comment by Koxiaet** [comment]
@bjorn3 Yes, and it happens for both the default `Cargo.toml` and other custom ones.

**Comment by flodiebold** [comment]
Are other features working? Do you e.g. get completions for `S`, or for standard library types?

**Comment by Koxiaet** [comment]
I'm not really sure what I'm _supposed_ to get, so I can't say. Certainly when I write something like `HashMap::` and then press `<C-n>` it doesn't come up with anything useful - just a bunch of random words like `self`, `is_some`, `deref`, `Target`, and words I've used in my doc comments.

I have this in my init.vim too:

```vim
" deoplete
Plug 'Shougo/deoplete.nvim', { 'do': ':UpdateRemotePlugins' }
let g:deoplete#enable_at_startup = 1

" neosnippet
Plug 'Shougo/neosnippet.vim'
Plug 'Shougo/neosnippet-snippets'

imap <C-k> <Plug>(neosnippet_expand_or_jump)
smap <C-k> <Plug>(neosnippet_expand_or_jump)
xmap <C-k> <Plug>(neosnippet_expand_target)
```
I'm really not too familiar with Language Servers, plugins, autocomplete and vimscript at all, so I'm just pasting in things from the Internet at the moment.

**Comment by flodiebold** [comment]
That sounds like rust-analyzer is not correctly set up, but I don't know anything about the vim setup.

> I'm not really sure what I'm supposed to get, so I can't say.

If you just type `HashMap::` without importing it first, I'd expect there to be no completions. If there are, they're coming from something else. If you type `std::collections::HashMap::`, there should be `from`, `new`, and three `with_` methods. If you type `S::`, you should get `f()` and `from`.

**Comment by Koxiaet** [comment]
@flodiebold For both `std::collections::HashMap::` and `HashMap::` the suggestions dont work.

![screenshot_2020-02-15-20:44:09](https://user-images.githubusercontent.com/38139193/74594925-90232d00-5033-11ea-8cee-e66cee0ce874.png)

**Comment by Koxiaet** [comment]
Yes, I ran `rustup add component rust-src` and now it works. Leaving this issue open, but changing title.

**Comment by matklad** [comment]
Yup, added to the docs. 

We also produce an error message for this case, which I think is surfaced in VS Code, but might be swallowed by VIM? if that's the case, we should raise issue on the vim's lsp support library.

**Comment by lnicola** [comment]
Fixed in #3185?
